src/kani-compiler/src/codegen_cprover_gotoc/codegen/place.rs

```diff
@@ -385,25 +399,42 @@ impl<'tcx> GotocCtx<'tcx> {
                     ty::Slice(..) => before.goto_expr.index(idxe),
                     _ => unreachable!("must index an array"),
                 };
-                ProjectedPlace::new(
+                Ok(ProjectedPlace::new(
                     expr,
                     typ,
                     before.fat_ptr_goto_expr,
                     before.fat_ptr_mir_typ,
                     self,
-                )
+                ))
             }
             ProjectionElem::ConstantIndex { offset, min_length, from_end } => {
-                self.codegen_constant_index(before, offset, min_length, from_end)
+                Ok(self.codegen_constant_index(before, offset, min_length, from_end))
             }
             // Best effort to codegen subslice projection.
-            // This is known to fail with a CBMC invariant violation
-            // in some cases. Full support to be added in
-            // https://github.com/model-checking/kani/issues/357
+            // Full support to be added in
+            // https://github.com/model-checking/kani/issues/707
             ProjectionElem::Subslice { from, to, from_end } => {
                 // https://rust-lang.github.io/rfcs/2359-subslice-pattern-syntax.html
                 match before.mir_typ().kind() {
-                    ty::Array(..) => unimplemented!(),
+                    ty::Array(ty, len) => {
```

Comment:
Should this be a separate PR? Or change the title of this PR?

Reply:
Makes sense to do in a separate PR. I took it out (along with the expected test that was added).